graphql-dgs/src/main/kotlin/com/netflix/graphql/dgs/DgsGraphQLConfigurationProperties.kt

```diff
@@ -30,23 +34,36 @@ import org.springframework.validation.annotation.Validated
 @Suppress("ConfigurationProperties")
 data class DgsGraphQLConfigurationProperties(
     /** Path to the GraphQL endpoint without trailing slash. */
-    @DefaultValue("/graphql") val path: String,
-    @DefaultValue val graphiql: DgsGraphiQLConfigurationProperties,
-    @DefaultValue val schemaJson: DgsSchemaJsonConfigurationProperties
+    @DefaultValue("/graphql") @field:ValidPath val path: String,
+    @DefaultValue @Valid val graphiql: DgsGraphiQLConfigurationProperties,
+    @DefaultValue @Valid val schemaJson: DgsSchemaJsonConfigurationProperties
 ) {
     /**
      * Configuration properties for GraphiQL.
      */
     data class DgsGraphiQLConfigurationProperties(
```

Comment:
There is one caveat on using `@Validated` on the Kotlin data class, it forces Spring to AOP it which fails as data class are final and not able to be opened. 

If we want to go w/ this approach we will need to also run `kotlin-allopen` or `kotlin-spring` compiler plugin. I know we are trying to use as few plugins as possible though.